src/main.cpp

```diff
@@ -22,6 +23,19 @@ constexpr float player_velocity_v{0.5f};
 bool playerUp{}, playerDown{}, playerLeft{}, playerRight{};
 Vec2<float> playerDirection{};
 
+struct InputSystem
+{
+  std::set<SDL_Keycode> pressedKeys;
+  std::set<SDL_Keycode> releasedKeys;
+  std::set<SDL_Keycode> heldKeys;
+
+  void reset()
+  {
+    this->pressedKeys.clear();
```

Comment:
Why isn't `heldKeys` cleared? Also, this function is mostly unnecessary, you can "clear" the entire object via `= {}`.